frontend/src/features/projects/dispose/tabs/EnhancedReferralTab.tsx

```diff
@@ -18,15 +17,13 @@ const EnhancedReferralTab: React.FunctionComponent<IEnhancedReferralTabProps> =
   setSubmitStatusCode,
   goToGreTransferred,
 }: IEnhancedReferralTabProps) => {
-  const { submitForm } = useFormikContext();
   return (
     <>
       <AgencyResponseForm isReadOnly={isReadOnly} />
       <EnhancedReferralCompleteForm
         isReadOnly={isReadOnly}
         onClickOnHold={() => {
           setSubmitStatusCode(ReviewWorkflowStatus.OnHold);
-          submitForm();
```

Comment:
calling this after the set... call above makes the assumption that the value submitStatusCode value will be set before submitForm() is executed. This is not guaranteed. I've refactored this so that the actions watch the value of submitStatusCode via useEffect and call submitForm if necessary.